### PR TITLE
Data locality

### PR DESCRIPTION
**Pull request by faddat**
remaking from notional-labs repo.

This PR is for stage 1 of 

* #548

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,38 +32,37 @@ tools:
...
-		go test $(LDFLAGS) -run=NOTEST -bench=RandomBytes . && \
-		go test $(LDFLAGS) -tags cleveldb,rocksdb,pebbledb -run=NOTEST -bench=Small . && \
-		go test $(LDFLAGS) -tags cleveldb,rocksdb,pebbledb -run=NOTEST -bench=Medium . && \
-		go test $(LDFLAGS) -tags cleveldb,rocksdb,pebbledb -run=NOTEST -timeout=30m -bench=Large . && \
-		go test $(LDFLAGS) -run=NOTEST -bench=Mem . && \
-		go test $(LDFLAGS) -run=NOTEST -timeout=60m -bench=LevelDB .
+		go test $(LDFLAGS) -bench=RandomBytes . && \
+		go test $(LDFLAGS) -tags boltdb,badgerdb,pebbledb -bench=Small . && \
+		go test $(LDFLAGS) -tags boltdb,badgerdb,pebbledb -bench=Medium . && \
+		go test $(LDFLAGS) -tags boltdb,badgerdb,pebbledb -timeout=30m -bench=Large . && \
```
Comment (robert-zaremba) [inline Makefile:63]: so, are we still using boltdb and badger db?

**Review by marbar3778 — COMMENTED** [review]
please break out the current changes into their own pr. this work needs to be specific to each pr not conflated with other work

File: Makefile [hunk 2]
```diff
@@ -129,5 +125,5 @@ proto-check-breaking:
 
 proto-gen:
 	@echo "Generating Protobuf files"
-	$(DOCKER) run --rm -v $(CURDIR):/workspace --workdir /workspace $(protoImageName) sh scripts/protocgen.sh
```
Comment (julienrbrt) [inline Makefile:132]: ditto

**Comment by faddat** [comment]
thanks!

**Comment by faddat** [comment]
Hey, sorry about that guys.  I began this branch from the branch for the earlier pr #554 

I'll close and re-open from main. 

I'm currently on vacation, but I've asked Tuan (@baabeetaa ) Khanh (catshark) and Vuong to have a look at this over the weekend.  BRB with a fresh one.

**Pull request closed without merging**